src/urllib3/connectionpool.py

```diff
@@ -150,22 +160,22 @@ class HTTPConnectionPool(ConnectionPool, RequestMethods):
     """
 
     scheme = "http"
-    ConnectionCls = HTTPConnection
+    ConnectionCls: Type[Union[HTTPConnection, HTTPSConnection]] = HTTPConnection
     ResponseCls = HTTPResponse
 
     def __init__(
         self,
-        host,
-        port=None,
-        timeout=Timeout.DEFAULT_TIMEOUT,
-        maxsize=1,
-        block=False,
-        headers=None,
-        retries=None,
-        _proxy=None,
-        _proxy_headers=None,
-        _proxy_config=None,
-        **conn_kw,
+        host: Optional[str],
```

Comment:
same as above, do we need Optional?

Reply:
Without `Optional`, we get
`src/urllib3/connectionpool.py:1036: error: Argument 1 to "HTTPConnectionPool" has incompatible type "Optional[str]"; expected "str"`